src/objectExplorer/objectExplorerService.ts

```diff
@@ -302,6 +306,20 @@ export class ObjectExplorerService {
         const result: TreeNodeInfo[] = [];
 
         let savedConnections = await this._connectionManager.connectionStore.readAllConnections();
+        // Remove any connections that have duplicated IDs
+        if (savedConnections.length > 0) {
+            const uniqueConnections = new Map<string, IConnectionProfileWithSource>();
+            for (const conn of savedConnections) {
+                if (!uniqueConnections.has(conn.id)) {
+                    uniqueConnections.set(conn.id, conn);
+                } else {
+                    this._logger.verbose(
+                        `Duplicate connection ID found: ${conn.id}. Removing duplicate connection.`,
+                    );
+                }
+            }
+            savedConnections = Array.from(uniqueConnections.values());
```

Comment:
I feel like this logic should be in the readAllConnections() method for universality instead of in OE.  What do you think?

Reply:
Yup, I agree. This is just a sanity check for OE  so we don't run into any duplicate id issues. Ideally, we should have this in `readAllConnections` with an error message surfaced to the users.